libafl_frida/build.rs

```diff
@@ -9,4 +9,17 @@ fn main() {
     // Force linking against libc++
     #[cfg(unix)]
     println!("cargo:rustc-link-lib=dylib=c++");
+
+    // Build the test harness
+    // clang++ -shared -fPIC -o harness.so harness.cpp
+    #[cfg(unix)]
+    std::process::Command::new("clang++")
```

Comment:
This should potentially use the `cc` crate to have (some) support for other environments

Reply:
I did not find a way to build a shared library with cc. If you have an idea - please let me know

Reply:
```
cc::build::new()
.flag("-shared")
.flag("-fPIC")
```
?

Reply:
Nope, this does not work. It produces a .a file and ignores the -shared. There is no linker running as well.
This is what I run:
`    cc::Build::new()
        .flag("-v")
        .flag("-shared")
        .flag("-fPIC")
        .flag("-O0")
        .file("test_harness.cpp")
        .compile("test_harness.so");
`
This is what it resulted in:
`file /home/me/LibAFL/target/release/build/libafl_frida-3e26f9fa0e88b5c2/out/libtest_harness.so.a 
/home/me/LibAFL/target/release/build/libafl_frida-3e26f9fa0e88b5c2/out/libtest_harness.so.a: current ar archive`

Reply:
Hmm, sad...

Alternatively maybe we can reuse the compiler detection from libafl_cc's build.rs here?
https://github.com/AFLplusplus/LibAFL/blob/3d126f21ccae72617c454ccc999e3ee93433b95c/libafl_cc/build.rs#L293

Assuming clang is in the path may or may not work, I fear.
Or worst case ignore the testcase if clang wasn't found?

Reply:
I could copy the code over, but am not sure it is worth the effort for the current initial state where a single platform is supported by Asan (and therefore for its test). The compiler detection in libafl_cc is quite complicated, and for UNIX eventually just checks if llvm-config is in the path and then runs it. I guess that if llvm-config is in the path, clang++ will be there as well. I can start with checking for clang presence and skip compilation if not found, then the test will be skipped if the harness is not found.

Reply:
Added checking for clang and failing the test if the harness not found

Reply:
It's a bit unfortunate `cargo test` can fail because of the PATH now, maybe we should make this optional somehow? Also, on MacOS the file extension would usually be dylib, right?
But anyway we can merge and then fix these things later